mozilla_django_oidc/auth.py

```diff
@@ -44,21 +45,27 @@ class OIDCAuthenticationBackend(ModelBackend):
 
     def __init__(self, *args, **kwargs):
         """Initialize settings."""
-        self.OIDC_OP_TOKEN_ENDPOINT = self.get_settings("OIDC_OP_TOKEN_ENDPOINT")
-        self.OIDC_OP_USER_ENDPOINT = self.get_settings("OIDC_OP_USER_ENDPOINT")
-        self.OIDC_OP_JWKS_ENDPOINT = self.get_settings("OIDC_OP_JWKS_ENDPOINT", None)
-        self.OIDC_RP_CLIENT_ID = self.get_settings("OIDC_RP_CLIENT_ID")
-        self.OIDC_RP_CLIENT_SECRET = self.get_settings("OIDC_RP_CLIENT_SECRET")
-        self.OIDC_RP_SIGN_ALGO = self.get_settings("OIDC_RP_SIGN_ALGO", "HS256")
-        self.OIDC_RP_IDP_SIGN_KEY = self.get_settings("OIDC_RP_IDP_SIGN_KEY", None)
-
-        if self.OIDC_RP_SIGN_ALGO.startswith("RS") and (
-            self.OIDC_RP_IDP_SIGN_KEY is None and self.OIDC_OP_JWKS_ENDPOINT is None
+        self.UserModel = get_user_model()
+
+    @cached_property
+    def settings(self):
+        retval = {
```

Comment:
@stevenbal is this going to break our library?

Reply:
With the way our library is implemented now, yes, because we rely on overriding `get_settings`

Reply:
I don't think this is too much of a problem? I don't see any changes to that method in particular and instead it will allow us to remove our `__init__` override since settings are now consumed lazily. So, what am I missing? :smile:

Reply:
Whoops, I skimmed over it too quickly and thought `get_settings` was removed here altogether :sweat_smile:. You're right, this should work and the init override can be removed.